pinot-core/src/main/java/com/linkedin/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -146,18 +147,24 @@ public DataTable processQuery(final QueryRequest queryRequest, ExecutorService e
       LOGGER.debug("Searching Instance for Request Id - {}, browse took: {}", instanceRequest.getRequestId(),
           queryProcessingTimer.getDurationNs());
       LOGGER.debug("InstanceResponse for Request Id - {} : {}", instanceRequest.getRequestId(), dataTable.toString());
-      dataTableMetadata
-          .put(DataTable.TIME_USED_MS_METADATA_KEY, Long.toString(queryProcessingTimer.getDurationMs()));
+      dataTableMetadata.put(DataTable.TIME_USED_MS_METADATA_KEY, Long.toString(queryProcessingTimer.getDurationMs()));
       dataTableMetadata.put(DataTable.REQUEST_ID_METADATA_KEY, Long.toString(instanceRequest.getRequestId()));
-      dataTableMetadata
-          .put(DataTable.TRACE_INFO_METADATA_KEY, TraceContext.getTraceInfoOfRequestId(instanceRequest.getRequestId()));
+      dataTableMetadata.put(DataTable.TRACE_INFO_METADATA_KEY,
+          TraceContext.getTraceInfoOfRequestId(instanceRequest.getRequestId()));
 
       // Update the total docs in the metadata based on un-pruned segments.
       dataTableMetadata.put(DataTable.TOTAL_DOCS_METADATA_KEY, String.valueOf(totalRawDocs));
       return dataTable;
     } catch (Exception e) {
       _serverMetrics.addMeteredQueryValue(instanceRequest.getQuery(), ServerMeter.QUERY_EXECUTION_EXCEPTIONS, 1);
-      LOGGER.error("Exception processing requestId {}", requestId, e);
+
+      // Do not log error for NumberFormatException because it's caused by bad query
+      if (e instanceof NumberFormatException) {
```

Comment:
May be better to add a new RuntimException called (say) BadQueryException.
If we explicitly raise this exception, then we can use it for multiple different errors